tdrs-backend/docker-compose.yml

```diff
@@ -10,6 +10,59 @@ services:
       - ./reports:/zap/wrk/:rw
       - ../scripts/zap-hook.py:/zap/scripts/zap-hook.py:ro
 
+  # Init container to set up Vault volume permissions
+  vault-init:
+    image: alpine:3.20
+    container_name: vault-init
+    user: "0:0"
+    restart: "no"
+    volumes:
+      - vault_data:/vault/data
+      - vault_logs:/vault/logs
+    command: ["sh", "-c", "chown -R 100:1000 /vault/data /vault/logs && chmod -R 755 /vault/data /vault/logs"]
+
+  
+
+  # HashiCorp Vault Community Edition
+  vault:
+    image: hashicorp/vault:latest
+    container_name: vault
+    depends_on:
+      - vault-init
```

Comment:
we can also consider to apply default credentials for local docker. I did not do that so everyone has to explorer how to add the creds in vault. that is something we all have to learn.